hazelcast-jet-core/src/main/java/com/hazelcast/jet/impl/JobCoordinationService.java

```diff
@@ -167,6 +180,20 @@ public void submitJob(long jobId, Data dag, JobConfig config) {
         nodeEngine.getExecutionService().execute(COORDINATOR_EXECUTOR_NAME, () -> tryStartJob(masterContext));
     }
 
+    private boolean hasActiveJobWithName(@Nonnull String jobName) {
+        // if scanJob() has not run yet, master context objects may not be initialized.
+        // in this case, we cannot check if the new job submission has a duplicate job name.
+        // therefore, we will retry until scanJob() task runs at least once.
+        if (!jobsScanned) {
+            throw new RetryableHazelcastException("Cannot submit job with name '" + jobName
```

Comment:
`scanJobs` doesn't create `MasterContext` for suspended jobs, we need to fix that.